lib/iris/_data_manager.py

```diff
@@ -390,7 +421,7 @@ def data(self, data):
         # Reset the fill-value appropriately.
         if ma.isMaskedArray(data):
             # Align the data manager fill-value with the numpy fill-value.
-            self.fill_value = data.fill_value
+            self._propagate_masked_data_fill_value()
```

Comment:
I also spotted that the `_propagate_masked_data_fill_value` was being called twice during testing ... which is really a bit undesirable, so I've addressed this in-line with my other changes :+1: